src/instance.rs

```diff
@@ -1215,6 +1219,35 @@ impl<T> Py<T> {
         unsafe { Py::from_borrowed_ptr(py, self.0.as_ptr()) }
     }
 
+    /// Drops `self` and immediately decreases its reference count.
+    ///
+    /// This method is a micro-optimisation over [`Drop`] if you happen to be holding the GIL
+    /// already.
```

Comment:
It could be worth mentioning that if you use `Bound` instead then there's no need for the explicit micro-optimisation, as it can assume the GIL is held.

Reply:
This is my first time seeing the new `Bound` API - what would you recommend writing here?

> Note that if you are using [`Bound`], you do not need to use [`Py::drop_ref`] since [`Bound`] guarantees that the GIL is held.

Reply:
That sounds great, yes 👍 

I'm yet to complete documentation, but I think the best way to describe it is `Bound<'py, T>` can be thought of a `(Py<T>, Python<'py>)` tuple, and because it has that GIL attachment it's possible to make lots of convenient assumptions about what can be done with it.

Reply:
Thanks for the positive feedback! I've literally spent years thinking about how to replace `&PyAny` with something with better CPU and memory performance without breaking everyone's code instantly. Thanks to a lot of help from many people's input I think the new "Bound" API is a solid improvement. Despite that, it will add complexity to users (by giving them full control over Python object lifetimes), and `Bound<'py, PyAny>` is definitely more awkward to read and write than `&PyAny`. So I'm equally excited and terrified to see what users think of the change...